### PR TITLE
Improve Activator.updateConstraints

### PR DESCRIPTION
**Pull request by gmlwhdtjd**
So far, we are deactivating all applied constraint and activating new constraint.

To improve ```Activator.updateConstraints(activation: Activation, constraints: [NSLayoutConstraint])```, 
it need to compare the applied constraint with the new constraint and only update the changes. So it can be used to compare custom hashValues ​​using ```NSLayoutConstraint``` values ​​like ```firstItem```, ```firstAttribute```, etc.

By the way, 
```WeakReference.Origin``` was already an object that required a ```CustomHashable```
So, Changes can be found using ```Set<WeakReference<NSLayoutConstraint>>``` without any special changes.

And then,```WeakReference``` was used only for ```NSLayoutConstraint```, so I clean up it as a ```WeakConstraint``` and changed it to a component of ```Activation```, not Util.

PS. Please also review whether there are any problems in terms of performance.

### CODE REVIEW

**Review by oozoofrog — APPROVED** [review]
- changing some naming in update constraints, LGTM
- removing WeakReference should be good to performance, also LGTM
- adding tests for WeakConstraint seems also may good. but my guess, you don't seem to need it this time.

File: Sources/SwiftLayout/Core/Activator/Component/WeakConstraint.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        var hasher = Hasher()
+        hasher.combine(origin?.firstItem as? NSObject)
+        hasher.combine(origin?.firstAttribute)
+        hasher.combine(origin?.secondItem as? NSObject)
+        hasher.combine(origin?.secondAttribute)
+        hasher.combine(origin?.relation)
+        hasher.combine(origin?.constant)
+        hasher.combine(origin?.multiplier)
+        hasher.combine(origin?.priority)
+        self.customHashValue = hasher.finalize()
```
Comment (oozoofrog) [inline Sources/SwiftLayout/Core/Activator/Component/WeakConstraint.swift:26]: very agreed 👍

**Review by oozoofrog — APPROVED** [review]

File: Sources/SwiftLayout/Core/Activator/Component/WeakConstraint.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    static func == (lhs: WeakConstraint, rhs: WeakConstraint) -> Bool {
+        lhs.hashValue == rhs.hashValue
+    }
+}
+
+extension Collection where Element: NSLayoutConstraint {
+    var weakens: [WeakConstraint] {
+        map(WeakConstraint.init)
+    }
+}
```
Comment (oozoofrog) [inline Sources/SwiftLayout/Core/Activator/Component/WeakConstraint.swift:44]: @gmlwhdtjd how about removing this? anytime at seeing this, feeling original sin.

Comment (gmlwhdtjd) [reply Sources/SwiftLayout/Core/Activator/Component/WeakConstraint.swift:44]: I currently have some uses, what do you want to do?

Comment (oozoofrog) [reply Sources/SwiftLayout/Core/Activator/Component/WeakConstraint.swift:44]: use map(WeakConstraint.init)

😄

Comment (oozoofrog) [reply Sources/SwiftLayout/Core/Activator/Component/WeakConstraint.swift:44]: oh it just ask your opinion. and not for this pr this time.

**Pull request merged**